gravitl/netmaker
# Gra 1198 enrollment keys

Do not merge till post 0.18.1
Testing steps:

- [x] Ensure you can create a enrollment keys only with admin privileges
- `curl -X POST -H "Authorization: Bearer <admin JWT>" -d '{"expiration":0,"uses_remaining":10,"networks":["networktoJoin1","network2"],"unlimited":false,"tags":[]}' https://<mydomain.com>/api/v1/enrollment-keys`
  `curl -H "Authorization: Bearer <admin JWT>" https://mydomain.com/api/v1/enrollment-keys | jq`
uses_remaining = how many uses are on it
expiration is set with unix time
unlimited indicates unlimited use one of those must be set, otherwise should fail

- 

- [x] Ensure you can get enrollment keys only with admin privileges

- `curl -H "Authorization: Bearer <admin JWT>" https://<mydomain.com>/api/v1/enrollment-keys | jq`
- [x] Ensure you can delete an enrollment key only with admin privilege
- `curl -X DELETE -H "Authorization: Bearer <admin JWT>" https://<mydomain.com>/api/v1/enrollment-keys/<keyValue>`

- [x] Check PR https://github.com/gravitl/netclient/pull/223 for netclient usability
